lib/google/gax/api_callable.rb

```diff
@@ -47,6 +50,8 @@ module Gax
     # @param func [Proc] used to make a bare rpc call
     # @param timeout [Numeric] client-side timeout for API calls
     # @param metadata [Hash]  request headers
+    # @param retry_codes [Array<Integer>] The error codes to retry
```

Comment:
Instead of passing these `retry_*` arguments as parameters here and forwarding them to a new `RetryManager` why not pass a `RetryManager` here directly and allow the user to customize via subclassing or by creating a new instance of the default `RetryManager` with non-default arguments?

Reply:
Ohhh, so give users the ability to completely control the retry mechanics, even user their own manager object if they want to? I like it! So object oriented!

We will still need a way to configure different client methods with different retry configuration. But I think we can do something here that will better address #165. Do you want that to be done in this PR or do you want that to be a follow up PR?

Reply:
I have added RetryPolicy, which can be configured. Or, users can provide a Proc (or any object that implements a `#call` method) to determine if the call should be retried, and implement the incremental backoff. PTAL.